### ISSUE TITLE
for (let x of y) generates code that doesn't work without a polyfil (symbols)

### ISSUE THREAD

**Issue by stefanpenner** [thoov/ember-weakmap#7]
https://github.com/thoov/ember-weakmap/blob/master/addon/weak-map.js#L21 produces:

``` js
     try {
        for (var _iterator = iterable[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
          var iteration = _step.value;

          var _iteration = _slicedToArray(iteration, 2);

          var key = _iteration[0];
          var value = _iteration[1];

          if (typeof iteration !== 'object') {
            throw new TypeError('Iterator value ' + key + ' is not an entry object');
          }

          this.set(key, value);
        }
      } catch (err) {
        _didIteratorError = true;
        _iteratorError = err;
      } finally {
        try {
          if (!_iteratorNormalCompletion && _iterator['return']) {
            _iterator['return']();
          }
        } finally {
          if (_didIteratorError) {
            throw _iteratorError;
          }
        }
      }
    }
```

and

``` js

  var _slicedToArray = (function () {
    function sliceIterator(arr, i) {
      var _arr = [];var _n = true;var _d = false;var _e = undefined;try {
        for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) {
          _arr.push(_s.value);if (i && _arr.length === i) break;
        }
      } catch (err) {
        _d = true;_e = err;
      } finally {
        try {
          if (!_n && _i['return']) _i['return']();
        } finally {
          if (_d) throw _e;
        }
      }return _arr;
    }return function (arr, i) {
      if (Array.isArray(arr)) {
        return arr;
      } else if (Symbol.iterator in Object(arr)) {
        return sliceIterator(arr, i);
      } else {
        throw new TypeError('Invalid attempt to destructure non-iterable instance');
      }
    };
  })();
```

**Comment by thoov** [comment]
@stefanpenner let me speed this up

**Comment by thoov** [comment]
@stefanpenner The [spec says](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap) that WeakMaps should take an iterable which would mean that Symbol is required. Do you suggest that we dont treat it as an iterable and do a simple loop over it?

**Comment by stefanpenner** [comment]
The perf is of less concern, rather that this enforces a Symbol polyfill that previously did not exist.

**Comment by stefanpenner** [comment]
> The perf is of less concern, rather that this enforces a Symbol polyfil that previously did not exist.

The issue i suppose is more that ^^ is a semver break.

**Comment by thoov** [comment]
@stefanpenner How do you suggest overriding ember-metals weakmap?

**Comment by thoov** [comment]
Also what is the status with Ember.WeakMap making an appearance?

**Comment by stefanpenner** [comment]
> Also what is the status with Ember.WeakMap making an appearance?

Good point, let me point more people to https://github.com/emberjs/rfcs/pull/91 so we can move that forward.

- [ ] get more 👀  on https://github.com/emberjs/rfcs/pull/91
- [ ] apply feedback
- [ ] FEATURE Flag the existing implementation
- [ ] make this the polyfil for backwards compat

**Comment by Turbo87** [comment]
@thoov I guess this can be closed now that #8 was merged?
